text/0000-format.md

```diff
@@ -0,0 +1,64 @@
+- Feature Name: format
+- Start Date: 2016-07-27
+- RFC PR: (leave this empty)
+- Pony Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes the seperation of the `string()` method and formatting. This wil remove formatting from the builtin package of the standard library and seperate the disperate concerns of producing a string representation of an object and formatting strings.
+
+# Motivation
+
+String formatting should not be involved in the string representation of an object. Pony's string formatting should also have improved documentation since it differs from that of most other languages.
+
+# Detailed design
+
+- Redefine the Stringable interface:
+```pony
+interface box Stringable
+  """
+  Things that can be turned into a String.
+  """
+  fun string(): String
+    """
+    Generate a string representation of this object.
+    """
+```
+- Modify all classes that currently implement the Stringable interface to reflect the new interface
+- Allow numbers to keep implementing the interface by modifying the `_ToString` primitive in builtin to create strings in base 10 only.
+- Create a package `format` that includes a `Format` primitive with the following methods:
+```pony
+primitive Format
+  fun apply(str: String, fmt: FormatSettings = FormatSettingsDefault): String =>
+  
+  fun int[A: (Int & Integer[A])](x: A,
+    fmt: FormatSettings[FormatInt, PrefixNumber] = FormatDefaultNumber
+  ): String =>
+  
+  fun float[A: (Float & FloatingPoint[A])](x: A,
+    fmt: FormatSettings[FormatFloat, PrefixNumber] = FormatDefaultNumber
+  ): String =>
```

Comment:
These should probably be `String iso` to give the caller more flexibility.  The strings are being built from scratch each time, so there is no loss on the receiver's end.